src/web/demo.rkt

```diff
@@ -309,6 +311,35 @@
      (redirect-to (add-prefix (format "~a.~a/graph.html" hash *herbie-commit*)) see-other))
    (url main)))
 
+;; (await fetch('/api/sample', {method: 'POST', body: JSON.stringify({formula: "(FPCore (x) (- (sqrt (+ x 1))))", seed: 5})})).json()
+(define (handle-sample-endpoint post-data)
+  (define formula (read-syntax 'web (open-input-string (hash-ref post-data `formula))))
+  (define seed (hash-ref post-data `seed))
+  (eprintf "Job started on ~a..." formula)
+
+  (define result (get-sample (parse-test formula) #:seed seed))
+
+  (eprintf " complete\n")
+  (hasheq
+    'points result))
+
+(define (call-api-fn req endpoint)
+  (define post-body (request-post-data/raw req))
+  (define post-data (cond (post-body (bytes->jsexpr post-body)) (#t #f)))
+  ;; use hash-ref to access values in the object, e.g.
+  ;; (hash-ref post-data 'key)
+  ;; test via Herbie demo page: (await fetch('/api/sample', {method: 'POST', body: JSON.stringify({a: 42})})).json()
+  (define json-obj (if (equal? endpoint "sample") (handle-sample-endpoint post-data) (hasheq
+                     'endpoint endpoint
+                     'data post-data
+                     )))
+  (response
+    200 #"OK"
+    (current-seconds) APPLICATION/JSON-MIME-TYPE
+    empty
+    (λ (op) (write-json json-obj op)))
```

Comment:
It looks like this always returns successfully, even if you pass an endpoint that doesn't exist? Why? I think it's probably better to do the routing inside `dispatch`, and if the goal is to abstract over JSON handling write a wrapper function for that.

Reply:
Yeah, that should probably 404 instead of echoing, will fix.

I'll move this stuff into `dispatch` as suggested.